component/loki/source/cloudflare/internal/cloudflaretarget/fields.go

```diff
@@ -41,18 +44,25 @@ var (
 	}...)
 )
 
-// Fields returns the mapping of FieldsType to the set of fields it represents.
-func Fields(t FieldsType) ([]string, error) {
+// Fields returns the union of a set of fields represented by the Fieldtype and the given additional fields. The returned slice will contain no duplicates.
+func Fields(t FieldsType, additionalFields []string) ([]string, error) {
+	var fields []string
 	switch t {
 	case FieldsTypeDefault:
-		return defaultFields, nil
+		fields = append(defaultFields, additionalFields...)
 	case FieldsTypeMinimal:
-		return minimalFields, nil
+		fields = append(minimalFields, additionalFields...)
 	case FieldsTypeExtended:
-		return extendedFields, nil
+		fields = append(extendedFields, additionalFields...)
 	case FieldsTypeAll:
-		return allFields, nil
+		fields = append(allFields, additionalFields...)
+	case FieldsTypeCustom:
+		fields = make([]string, len(additionalFields))
+		copy(fields, additionalFields)
```

Comment:
You can make this into one line with
```suggestion
		fields = append(fields, additionalFields)
```
I would also be fine with returning just the `additionalFields` since it shouldn't be modified anywhere.

Reply:
still nice to clean up the duplicates. If a user has a long list of additional fields it could happen that they define twice the same field in the list by mistake